### PR TITLE
Refactor emails to use JSX as templating engine

### LINKED ISSUE #2914: RFC: Email Revamp
## Problem 

As our templates for emails are increasing, we're in the need to come up with a more sustainable way to handle them. Some of the main problems are:

- Lack of composability
- Lack of ability to visualize previews
- Lack of standardization
- Code feels convoluted
- Difficulty to enforce visual consistency

## Possible solutions

- Using a template engine like Nunjucks, EJS, PUG, etc.
- A full fledged library like [Maizzle](https://maizzle.com/docs/introduction/)
- Using JSX to generate static email-compatible HTML

## Proposal

Since all @calcom/core team is well versed on JSX, it is possible to keep using JSX to take advantage of React components to keep our code tidy and organized. Having our templates in JSK then can be rendered to simple static HTML using the [`renderToStaticMarkup`](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup) method.

This code can live in a new package in `packages/emails` called `@calcom/emails`.

First step would be to migrate our current templating to JSX as it is. Then we can ask @Jaibles to come up with a new design system. If we end up using Maizzle we can use Tailwind utilities to ease visual development.

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Migrates all email code to `packages/emails`
- Added new packages `@calcom/emails`
- Uses JSX to render emails instead of string templates
- Removes duplicate email code from "Wipe my Cal" app and uses it from `@calcom/emails` instead
- Removes duplicate email code from "Vital" app and uses it from `@calcom/emails` instead
- Closes #2914 

Migrated to JSX: 
- attendee-awaiting-payment-email.ts
- attendee-cancelled-email.ts
- attendee-declined-email.ts
- attendee-location-change-email.ts
- attendee-request-email.ts
- attendee-rescheduled-email.ts
- attendee-scheduled-email.ts
- organizer-cancelled-email.ts
- organizer-request-reschedule-email.ts
- organizer-rescheduled-email.ts
- organizer-scheduled-email.ts
- team-invite-email.ts
- feedback-email.ts
- forgot-password-email.ts
- organizer-location-change-email.ts
- attendee-request-reschedule-email.ts
- organizer-payment-refund-failed-email.ts
- organizer-request-email.ts
- organizer-request-reminder-email.ts

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] `yarn dev`
- [ ] Go to `http://localhost:3000/api/email`

### CODE REVIEW

File: packages/emails/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@calcom/emails",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {},
+  "dependencies": {
+    "@maizzle/framework": "^3.7.3"
```
Comment (zomars) [inline packages/emails/package.json:7]: We will use this to inject tailwind classname later on.

File: packages/emails/src/renderEmail.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import * as templates from "./templates";
+
+function renderEmail<K extends keyof typeof templates>(
+  template: K,
+  props: ComponentProps<typeof templates[K]>
+) {
+  const Component = templates[template];
+  return (
+    ReactDOMServer.renderToStaticMarkup(Component(props))
```
Comment (zomars) [inline packages/emails/src/renderEmail.ts:12]: This is where the magic happens

File: packages/emails/src/renderEmail.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      // Remove `<RawHtml />` injected scripts
+      .replace(/<script><\/script>/g, "")
+      .replace(
+        "<html>",
+        `<html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">`
+      )
```
Comment (zomars) [inline packages/emails/src/renderEmail.ts:18]: Injects some unsupported JSX attributes back to the HTML

**Review by zomars — COMMENTED** [review]
Ready for review

**Comment by hariombalhara** [comment]
@zomars Do you think we should test each and every email as it is a critical change

**Review by agustif — APPROVED** [review]
Super impressive work @zomars, Lots of smart workarounds in this PR! these changes make our Emails much more readable, and hopefully it will make them more robust against bugs and unintended side effects, now that we have better legibility!

✅ 
The emails also look great too
image

File: apps/web/pages/api/email.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
Comment (leog) [inline apps/web/pages/api/email.ts:1]: was about to ask if the hardcoded info in this endpoint were supposed to be hardcoded until I found the Development part of this in the README. Perhaps we can clarify in the implementation that this endpoint is for testing purposes only, or create another folder to have development-only endpoints

**Review by leog — APPROVED** [review]
Magnificent work @zomars. I saw that Tailwind per se was not really used, lots of inline styles, but the focus of componentizing the UI with JSX looks very neat and maintainable which I think is what was all about. Maybe another option for future follow-ups can be using https://github.com/ben-rogerson/twin.macro, worked very well in my last job.

File: apps/web/pages/api/email.ts (hunk 4 above)
Comment (zomars) [reply apps/web/pages/api/email.ts:1]: We can safely remove it or make it work only on dev environments.

Comment (zomars) [reply apps/web/pages/api/email.ts:1]: also I have my eye on Maizzle to inline tailwind styles later on.

**Pull request merged**